### PR TITLE
Jackie - Hotel - Ampers

### PR DESCRIPTION
**Pull request by Jackiesan**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | When designing the class Block, I was considering having Block inherit from Reservation. At first, I thought that I would save a lot of time if I pass most of the behaviors of reservation to Block. As I started writing more pseudocode I realized that Block and Reservation contain similar information but their behaviors are very different. Realizing that Block *is* not a reservation but *has* reservations helped me realize that it should be it's own separate class. 
Describe a concept that you gained more clarity on as you worked on this assignment. | I gained more clarity on private methods. Private methods can be very useful if it returns information that does not have to accessed by other classes. 
Describe a _nominal test_ that you wrote for this assignment. | For the administrator class I wrote a test for the create block method in which the collection of the block's rooms has to equal to the number of rooms passed in the argument. 
Describe an _edge case test_ that you wrote for this assignment. | An edge, also for the same class and method (class Administrator, method create_block), is to return an argument error if no rooms are available for a block during that specific date range. These rooms are unavailable due to regular reservations or rooms saved for a block.
How do you feel you did in writing pseudocode first, then writing the tests and then the code? | I think I did a good job in writing pseudocode first. It definitely made the whole process of design easier. I also did a good job in writing the tests first as usual, but I found it challenging to write the tests first for the Block class since I needed to spend more time understanding what behaviors Block should have.

### CODE REVIEW

File: lib/administrator.rb [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+    def find_room(room_id)
+      check_room_id(room_id)
+      rooms.find{ |room| room.room_id == room_id}
+    end
+
+    def reservations_on_date(date)
+      check_date(date)
+
+      reservations.select { |reservation| (reservation.check_out_date == date) ||  (reservation.block_of_dates.include?(date) == true) }
```
Comment (CheezItMan) [inline lib/administrator.rb:38]: Question:  If the room is being checked out of on a given date, can other people check in on that date, usually yes.

**Comment by CheezItMan** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility |  Your `Administrator` class is doing an awful lot including telling if a reservation conflicts with a date.  That would go better in the `Reservation` class.  
Demonstrated loose coupling |  Your code is **very** tightly coupled.  `Reservation` and `Room` know about each other, and `Block` knows about `Reservation` and `Administrator` knows about'em all.  
Methods demonstrate a good use of encapsulation, inputs and outputs | Check
Wave 1 requirements |  Check
Wave 2 requirements |  Check
Wave 3 requirements |  Check
Summary | Overall pretty good, but you're `Reservation` class is a bit anemic.  Plus your `Room` class is doing a lot, I don't think it needs to do.  There is a lot of good here however.  Nice work!  Checkout our [reference implementation](https://github.com/droberts-ada/hotel/tree/dpr/solution)
